### PR TITLE
chore (anomaly detection): create proxy API endpoint

### PR DESCRIPTION
**Pull request by mifu67**
Create a proxy API endpoint for us to query the backend about anomalies. `get_historical_anomaly_data_from_seer` returns a placeholder.

### CODE REVIEW

File: src/sentry/seer/anomaly_detection/get_historical_anomalies.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+from sentry.incidents.models.alert_rule import AlertRule, AlertRuleStatus
+from sentry.models.project import Project
+from sentry.seer.anomaly_detection.types import AnomalyType
+
+
+def get_historical_anomaly_data_from_seer(
+    alert_rule: AlertRule, project: Project, start_string: str, end_string: str
+) -> list | None:
+    """
+    Send time series data to Seer and return anomaly detection response (PLACEHOLDER).
```
Comment (nhsiehgit) [inline src/sentry/seer/anomaly_detection/get_historical_anomalies.py:10]: Oh erm.

this is gonna be in a separate PR right?

File: src/sentry/seer/anomaly_detection/store_data.py [hunk 2]
```diff
@@ -1,32 +1,30 @@
...
 from sentry.net.http import connection_from_url
 from sentry.search.events.types import SnubaParams
 from sentry.seer.anomaly_detection.types import (
     AlertInSeer,
     AnomalyDetectionConfig,
     StoreDataRequest,
-    TimeSeriesPoint,
 )
-from sentry.seer.anomaly_detection.utils import translate_direction
+from sentry.seer.anomaly_detection.utils import (
```
Comment (nhsiehgit) [inline src/sentry/seer/anomaly_detection/store_data.py:19]: This refactor could probably be broken out into a separate PR 🤷

File: tests/sentry/incidents/endpoints/test_organization_alert_rule_anomalies.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
+from datetime import timedelta
+from unittest.mock import patch
+
+# import orjson
+import pytest
+from urllib3.response import HTTPResponse
+
+# from sentry.conf.server import SEER_ANOMALY_DETECTION_ENDPOINT_URL
```
Comment (nhsiehgit) [inline tests/sentry/incidents/endpoints/test_organization_alert_rule_anomalies.py:8]: delete?

File: tests/sentry/incidents/endpoints/test_organization_alert_rule_anomalies.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
+from datetime import timedelta
+from unittest.mock import patch
+
+# import orjson
```
Comment (nhsiehgit) [inline tests/sentry/incidents/endpoints/test_organization_alert_rule_anomalies.py:4]: delete?

File: tests/sentry/incidents/endpoints/test_organization_alert_rule_anomalies.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    endpoint = "sentry-api-0-organization-alert-rule-anomalies"
+
+    method = "get"
+
+    @with_feature("organizations:anomaly-detection-alerts")
+    @with_feature("organizations:incidents")
+    @patch(
+        "sentry.seer.anomaly_detection.store_data.seer_anomaly_detection_connection_pool.urlopen"
+    )
+    # @patch(
```
Comment (nhsiehgit) [inline tests/sentry/incidents/endpoints/test_organization_alert_rule_anomalies.py:37]: delete?

File: tests/sentry/incidents/endpoints/test_organization_alert_rule_anomalies.py [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+        alert_rule = self.create_alert_rule(
+            time_window=15,
+            sensitivity=AlertRuleSensitivity.MEDIUM,
+            seasonality=AlertRuleSeasonality.AUTO,
+            detection_type=AlertRuleDetectionType.DYNAMIC,
+        )
+
+        self.login_as(self.user)
+
+        # seer_return_value = {
```
Comment (nhsiehgit) [inline tests/sentry/incidents/endpoints/test_organization_alert_rule_anomalies.py:76]: delete?

**Review by nhsiehgit — APPROVED** [review]
bunch of non blocking nits

File: src/sentry/incidents/endpoints/organization_alert_rule_anomalies.py [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        examples=MetricAlertExamples.GET_METRIC_ALERT_ANOMALIES,
+    )
+    def get(self, request: Request, organization: Organization, alert_rule: AlertRule) -> Response:
+        """
+        Return a list of anomalies for a metric alert rule.
+        """
+        project = alert_rule.projects.first()
+        start = request.GET.get("start", None)
+        end = request.GET.get("end", None)
+
```
Comment (ceorourke) [inline src/sentry/incidents/endpoints/organization_alert_rule_anomalies.py:61]: should there be a feature flag check somewhere in here?

File: tests/sentry/incidents/endpoints/test_organization_alert_rule_anomalies.py (hunk 5 above)
Comment (ceorourke) [inline tests/sentry/incidents/endpoints/test_organization_alert_rule_anomalies.py:37]: is all the commented out stuff for the next pr?

Comment (mifu67) [reply tests/sentry/incidents/endpoints/test_organization_alert_rule_anomalies.py:37]: Yes it is, I figured since I'm planning to merge the next one in quick succession I would leave the comments in but I'll take them out for better style

Comment (nhsiehgit) [reply tests/sentry/incidents/endpoints/test_organization_alert_rule_anomalies.py:37]: :nod:

Yea - ferge asked us to split the previous PR up 😅 

probably not necessary since we have context, but i suppose its good practice lol

File: src/sentry/incidents/endpoints/organization_alert_rule_anomalies.py [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+        if not project or start is None or end is None:
+            return Response(
+                "Unable to get historical anomaly data: missing required argument(s) project, start, and/or end",
+                status=400,
+            )
+
+        anomalies = get_historical_anomaly_data_from_seer(alert_rule, project, start, end)
+        if anomalies is None:
+            return Response("Unable to get historical anomaly data", status=400)
```
[the remark this answers is not in the document]
Comment (mifu67) [reply src/sentry/incidents/endpoints/organization_alert_rule_anomalies.py:70]: If there are fewer than seven days of event data, return an empty array. Otherwise, we get an array of labeled "anomalies" (these are potential anomalies that have been evaluated by Seer)
